src/views/connectionUI.ts

```diff
@@ -84,19 +85,32 @@ export class ConnectionUI {
 	 * @param ignoreFocusOut Whether to ignoreFocusOut on the quickpick prompt
 	 * @returns The connection picked or created.
 	 */
-	public async promptForConnection(ignoreFocusOut = false): Promise<IConnectionInfo | undefined> {
+	public async promptForConnection(ignoreFocusOut: boolean = false): Promise<IConnectionInfo | undefined> {
 		let picklist = this._connectionStore.getPickListItems();
 		// We have recent connections - show them in a picklist
-		const selection = await this.promptItemChoice({
-			placeHolder: LocalizedConstants.recentConnectionsPlaceholder,
-			matchOnDescription: true,
-			ignoreFocusOut: ignoreFocusOut
-		}, picklist);
-		if (selection) {
-			return this.handleSelectedConnection(selection);
-		} else {
+		const connectionProfileQuickPick = await vscode.window.createQuickPick<IConnectionCredentialsQuickPickItem>();
+		connectionProfileQuickPick.items = picklist;
+		connectionProfileQuickPick.placeholder = LocalizedConstants.recentConnectionsPlaceholder;
+		connectionProfileQuickPick.matchOnDescription = true;
+		connectionProfileQuickPick.ignoreFocusOut = ignoreFocusOut;
+		connectionProfileQuickPick.canSelectMany = false;
+		connectionProfileQuickPick.busy = false;
+		connectionProfileQuickPick.show();
+		connectionProfileQuickPick.onDidChangeSelection(selection => {
+			if (selection[0]) {
+				// add progress notification and hide quickpick after user chooses an item from the quickpick
+				connectionProfileQuickPick.busy = true;
+				connectionProfileQuickPick.hide();
+				return this.handleSelectedConnection(selection[0]);
```

Comment:
I'm assuming this is meant to be returned from `promptForConnection`? That isn't going to work currently, the return is for the `onDidChangeSelection` callback. 

Right now this function is always going to return undefined. You need to use a deferred promise (or wrap the whole thing in a promise) to be able to handle this correctly.